ocaml/parsing/lexer.mll

```diff
@@ -128,6 +128,97 @@ let is_in_string = ref false
 let in_string () = !is_in_string
 let print_warnings = ref true
 
+type deferred_token =
+  { token : token
+  ; start_pos : Lexing.position
+  ; end_pos : Lexing.position
+  }
+
+let deferred_tokens : deferred_token Queue.t = Queue.create ()
```

Comment:
I think this queue will only ever have 0 or 1 things in it.  It's fine to use a queue for the convenient interface, but maybe note that fact in one of your comments for clarity.